x-pack/plugins/apm/server/lib/service_map/get_service_map_service_node_info.ts

```diff
@@ -43,21 +42,14 @@ export async function getServiceMapServiceNodeInfo({
 }: Options & { serviceName: string; environment?: string }) {
   const { start, end } = setup;
 
-  const environmentNotDefinedFilter = {
-    bool: { must_not: [{ exists: { field: SERVICE_ENVIRONMENT } }] },
-  };
-
   const filter: ESFilter[] = [
     { range: rangeFilter(start, end) },
     { term: { [SERVICE_NAME]: serviceName } },
   ];
 
-  if (environment) {
-    filter.push(
-      environment === ENVIRONMENT_NOT_DEFINED
-        ? environmentNotDefinedFilter
-        : { term: { [SERVICE_ENVIRONMENT]: environment } }
-    );
+  const environmentFilter = getEnvironmentUiFilterES(environment);
+  if (environmentFilter) {
+    filter.push(environmentFilter);
   }
```

Comment:
Slightly cleaner:

```suggestion
  const filter: ESFilter[] = [
    { range: rangeFilter(start, end) },
    { term: { [SERVICE_NAME]: serviceName } },
    ...[getEnvironmentUiFilterES(environment)]
  ];
```

Reply:
I tried this at first, but it fails the type check since `getEnvironmentUiFilterES` returns `ESFilter | undefined`. I could change it to return `ESFilter[]`, so we can always safely spread it in (`...getEnvironmentUiFilterES(environment)`).

Reply:
done in https://github.com/elastic/kibana/pull/71229/commits/4bf17eefcead1809e6ce987ed10be83b3eb6d169